deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -2041,57 +1981,43 @@ extension ChatViewController {
             previewProvider: nil,
             actionProvider: { [weak self] _ in
                 guard let self else { return nil }
-
                 let message = dcContext.getMessage(id: messageId)
-
                 var children: [UIMenuElement] = []
 
-                let showReaction = message.isInfo == false && message.isSetupMessage == false && message.type != DC_MSG_VIDEOCHAT_INVITATION && dcChat.canSend
-                if showReaction {
-                    children.append(reactionsMenu(indexPath: indexPath))
-                }
-
-                let replyMenuChildren: [UIMenuElement]
-                let messageIsFromMe = message.isFromCurrentSender
-
-                if message.isInfo {
-                    replyMenuChildren = []
-                } else if dcChat.canSend && self.isGroupChat && messageIsFromMe == false {
-                    replyMenuChildren = [
-                        UIAction.menuAction(localizationKey: "notify_reply_button", systemImageName: "arrowshape.turn.up.left.fill", indexPath: indexPath, action: { self.reply(at: $0 ) }),
-                        UIAction.menuAction(localizationKey: "reply_privately", systemImageName: "arrowshape.turn.up.left", indexPath: indexPath, action: { self.replyPrivatelyToMessage(at: $0 ) }),
-                    ]
-                } else if self.isGroupChat && messageIsFromMe == false {
-                    replyMenuChildren = [
-                        UIAction.menuAction(localizationKey: "reply_privately", systemImageName: "arrowshape.turn.up.left", indexPath: indexPath, action: { self.replyPrivatelyToMessage(at: $0 ) }),
-                    ]
-                } else if dcChat.canSend {
-                    replyMenuChildren = [
-                        UIAction.menuAction(localizationKey: "notify_reply_button", systemImageName: "arrowshape.turn.up.left.fill", indexPath: indexPath, action: { self.reply(at: $0 ) }),
-                    ]
-                } else {
-                    replyMenuChildren = []
+                if canReply(message: message) {
+                    if #available(iOS 16.0, *) {
+                        // together with preferredElementSize below, this shows the reaction options in a row
+                        reactionsMenuItems(appendTo: &children, indexPath: indexPath)
+                    } else {
```

Comment:
This is why we refactored this, right? To make it easier to use new APIs. It's good to see it working :)

Reply:
yes. and sth. as `inout` is the pattern that seems useful for that - otherwise, we would copy items around twice, including additional code. i am not too afraid on that :) but maybe we can change the function name to make that aspect clearer `appendReactionItems(to: ..., indexPath: ...)`